frontend/webapp/hooks/actions/useActionFormData.ts

```diff
@@ -43,10 +44,42 @@ export function useActionFormData() {
     return ok;
   };
 
+  const loadFormWithDrawerItem = (drawerItem: DrawerBaseItem) => {
```

Comment:
build update action locally and set the state once

```suggestion
const loadFormWithDrawerItem = (drawerItem: DrawerBaseItem) => {
  const { type, spec } = drawerItem.item as ActionDataParsed;
  
  const updatedData: ActionInput = {
    ...INITIAL,
    type,
  };

  Object.entries(spec).forEach(([k, v]) => {
    switch (k) {
      case 'actionName':
        updatedData.name = v as string;
        break;

      case 'disabled':
        updatedData.disable = v as boolean;
        break;

      case 'notes':
      case 'signals':
        updatedData[k] = v;
        break;

      default:
        updatedData.details = JSON.stringify({ [k]: v });
        break;
    }
  });

  // Set the accumulated changes in a single state update
  setFormData(updatedData);
};
```